File: 2021/day02/java/part1.java
import java.util.*;
import java.io.*;

public class part1 {

	public static void main(String[] args) throws FileNotFoundException{
		String fname = args[0];
		File fo = new File(fname);
		Scanner scanner = new Scanner(fo);
		ArrayList<String> lines = new ArrayList<String>();
		while(scanner.hasNextLine()){
			lines.add(scanner.nextLine());
		}

		System.out.println(solve(lines));
	}

	public static int solve(List<String> input){
		HashMap<String, Integer> m = new HashMap<String, Integer>();
		m.put("forward", 0);
		m.put("up", 0);
		m.put("down", 0);

		for (String line : input){
			String[] l = line.split(" ");
			m.put(l[0], m.get(l[0]) + Integer.parseInt(l[1]));
		}

		return m.get("forward") * (m.get("down") - m.get("up"));
	}
}

File: 2021/day02/python/part1.py
#!usr/bin/env python

import sys

def main(lines):
    m = {
        'forward': 0,
        'up': 0,
        'down': 0
    }
    for line in lines:
        line = line.split()
        m[line[0]] = m[line[0]] + int(line[1])

    return m['forward'] * (m['down'] - m['up'])


if __name__ == '__main__':
    fo = open(sys.argv[1], 'r')
    lines = fo.readlines()
    print(main(lines))
    fo.close()
